### PR TITLE
Add support of CAP40 ed25519SignedPayload signer for SetOptionOp

### LINKED ISSUE #541: Add support of CAP40 ed25519SignedPayload signer for SetOptionOp
Currently (Stellar Base v8.1.0) `SetOptionOp` does not support new CAP40 `ed25519SignedPayload` signer. This blocks creation and results in incorrect parsing of a transaction containing such an operation.

### PR DESCRIPTION
**Pull request by orbitlens**
Fix #541

### CODE REVIEW

File: src/operations/set_options.js [hunk 1]
```diff
@@ -41,6 +41,7 @@ function weightCheckFunction(value, name) {
  * @param {string} [opts.signer.ed25519PublicKey] - The ed25519 public key of the signer.
  * @param {Buffer|string} [opts.signer.sha256Hash] - sha256 hash (Buffer or hex string) of preimage that will unlock funds. Preimage should be used as signature of future transaction.
  * @param {Buffer|string} [opts.signer.preAuthTx] - Hash (Buffer or hex string) of transaction that will unlock funds.
+ * @param {string} [opts.signer.ed25519SignedPayload] - Signed payload signer (ed25519 public key + raw payload) for atomic transaction signature disclosure.
```
Comment (overcat) [inline src/operations/set_options.js:44]: Maybe we need to add corresponding changes in `index.d.ts`.

Comment (orbitlens) [reply src/operations/set_options.js:44]: Good point, thanks!
Updated typescript bindings in f4651697bb91bfbfe33add452bda50adaad1a0bb

**Comment by orbitlens** [comment]
@Shaptic could you please review?

**Review by Shaptic — APPROVED** [review]
Seems good to get this in now and add fancy things on top later; sorry for the lengthy delay in approval.

**Comment by orbitlens** [comment]
Thanks 👍

**Comment by Shaptic** [comment]
This is released in stellar-base and stellar-sdk now!

**Pull request merged**